### PR TITLE
APPS-991-document-reuse

### PR DESCRIPTION
**Pull request by mhrvol**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ dxCompiler uses [wdlTools](https://github.com/dnanexus/wdlTools), a parser that
   * All task and workflow names must be unique across the entire import tree
     * For example, if `A.wdl` imports `B.wdl` and `A.wdl` defines `workflow foo`, then `B.wdl` could not have a workflow or task named `foo`
```
Comment (commandlinegirl) [inline README.md]:
```suggestion
    * For example, if `A.wdl` imports `B.wdl` and `A.wdl` defines workflow `foo`, then `B.wdl` could not have a workflow or task named `foo`
```
```suggestion
    * For example, if `A.wdl` imports `B.wdl` and `A.wdl` defines `workflow foo`, then `B.wdl` could not have a workflow or task named `foo`
```

File: README.md [hunk 2]
```diff
@@ -113,6 +113,7 @@ dxCompiler uses [wdlTools](https://github.com/dnanexus/wdlTools), a parser that
...
   * The `Directory` type is only supported in stand-alone tasks, not yet in workflows
+  * Compilation of subworkflows(subtasks) used from different workflow files will create new workflows(applets), thus reuse is not available between built workflows.
```
Comment (commandlinegirl) [inline README.md]: Let's discuss it in person how we can make it clearer.

File: README.md [hunk 3]
```diff
@@ -111,8 +111,9 @@ dxCompiler uses [wdlTools](https://github.com/dnanexus/wdlTools), a parser that
 * WDL and CWL
   * Calls with missing arguments have [limited support](doc/ExpertOptions.md#task-and-workflow-inputs).
   * All task and workflow names must be unique across the entire import tree
-    * For example, if `A.wdl` imports `B.wdl` and `A.wdl` defines `workflow foo`, then `B.wdl` could not have a workflow or task named `foo`
+    * For example, if `A.wdl` imports `B.wdl` and `A.wdl` defines workflow `foo`, then `B.wdl` could not have a workflow or task named `foo`
   * The `Directory` type is only supported in stand-alone tasks, not yet in workflows
+  * Compilation of subworkflows(subtasks) used from different workflow files will create new workflows(applets), thus reuse is not available between built workflows.
```
Comment (emiloslavsky) [inline README.md]: Need to document the fact that DNAnexus subworkflows rely on the top-level workflow context and explicitly mention the dependencies of the DNAnexus subworkflow on the top-level workflow

**Review by emiloslavsky — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -131,7 +131,8 @@ To compile a CWL workflow/tool, you will first need to process it using the foll
 * WDL and CWL
   * Calls with missing arguments have [limited support](doc/ExpertOptions.md#task-and-workflow-inputs).
   * All task and workflow names must be unique across the entire import tree
-    * For example, if `A.wdl` imports `B.wdl` and `A.wdl` defines `workflow foo`, then `B.wdl` could not have a workflow or task named `foo`
+    * For example, if `A.wdl` imports `B.wdl` and `A.wdl` defines workflow `foo`, then `B.wdl` could not have a workflow or task named `foo`
```
Comment (commandlinegirl) [inline README.md]: could not -> cannot

**Review by commandlinegirl — DISMISSED** [review]
LGTM, with a minor change suggestion

**Review by commandlinegirl — APPROVED** [review]

**Pull request merged**